### PR TITLE
fix(snackbar): Apply visibility hidden when at rest. fixes #1258

### LINKED ISSUE #1258: Possible to see snackbars in hidden state on any platform with rubber-band scrolling at edges
### What MDC-Web Version are you using?

0.20.0

### What browser(s) is this bug affecting?

Chrome or Safari on OS X

### What are the steps to reproduce the bug?

1. Go to http://material-components-web.appspot.com/snackbar.html
2. Scroll to the bottom and continue scrolling to cause the browser to rubber-band past the end of the screen

### What is the expected behavior?

It shouldn't be possible to see the hidden snackbars.

### What is the actual behavior?

The boxes from the "hidden" snackbars are visible.

image

### Any other information you believe would be useful?

Presumably we're leaving these elements in a convenient resting position to transition in. Possible solutions:

* Switch to using animations and set up keyframes so that the resting place is further offscreen but it immediately moves to the current resting place at the beginning of the animation
* Since this is a JS component, programmatically update the transition style (or apply another class) before becoming active / after the transition ends after becoming inactive

### PR DESCRIPTION
**Pull request by rpatil26**

### CODE REVIEW

**Comment by rpatil26** [comment]
@amsheehan Can you please check this?

File: packages/mdc-snackbar/mdc-snackbar.scss [hunk 1]
```diff
@@ -69,6 +69,10 @@
     }
   }
 
+  &[aria-hidden] {
```
Comment (amsheehan) [inline packages/mdc-snackbar/mdc-snackbar.scss]: Why not select on `.mdc-snackbar` and `.mdc-snackbar--active` for this?

**Comment by rpatil26** [comment]
Thanks @amsheehan for the suggestion.

I had tried that already, but show or hide animation was getting messed up. To keep it sync, we need to apply CSS transition (on opacity may be because visibility animation is little tricky) and make sure the timing is in sync. Alternately we can also use JS and wait for transition end to alter the visibility. I feel it is increasing code size and introducing unnecessary complexity. Currently aria hidden is on transition end already and to me that looked the right place to hook it.

Trying to understand your rationale and thoughts, why the current one might not be a good idea?

**Comment by rpatil26** [comment]
Thanks Alex. Yes, it's good to specifically target aria-hidden `true` to make it work reliably. Done.

**Comment by rpatil26** [comment]
Hi @amsheehan, can you please merge it if it looks good now.

File: packages/mdc-snackbar/mdc-snackbar.scss (hunk 1 above)
Comment (rpatil26) [reply packages/mdc-snackbar/mdc-snackbar.scss]: Done!

Comment (rpatil26) [reply packages/mdc-snackbar/mdc-snackbar.scss]: Changes done as discussed!

**Pull request closed without merging**